ElfUtils/LinuxMap.cpp

```diff
@@ -21,14 +21,45 @@ namespace orbit_elf_utils {
 using orbit_elf_utils::ElfFile;
 using orbit_grpc_protos::ModuleInfo;
 
-static ErrorMessageOr<uint64_t> FileSize(const std::string& file_path) {
-  struct stat stat_buf {};
-  int ret = stat(file_path.c_str(), &stat_buf);
-  if (ret != 0) {
-    return ErrorMessage(absl::StrFormat("Unable to call stat with file \"%s\": %s", file_path,
-                                        SafeStrerror(errno)));
+ErrorMessageOr<ModuleInfo> CreateModule(const std::filesystem::path& module_path,
+                                        uint64_t start_address, uint64_t end_address) {
+  // This excludes mapped character or block devices.
+  if (absl::StartsWith(module_path.string(), "/dev/")) {
+    return ErrorMessage(absl::StrFormat(
+        "The module \"%s\" is a character or block device (is in /dev/)", module_path));
   }
-  return stat_buf.st_size;
+
+  if (!std::filesystem::exists(module_path)) {
+    return ErrorMessage(absl::StrFormat("The module file \"%s\" does not exist", module_path));
+  }
+  std::error_code error;
+  uint64_t file_size = std::filesystem::file_size(module_path, error);
+  if (error) {
+    return ErrorMessage(
+        absl::StrFormat("Unable to get size of \"%s\": %s", module_path, error.message()));
+  }
+
+  ErrorMessageOr<std::unique_ptr<ElfFile>> elf_file = ElfFile::Create(module_path);
+  if (!elf_file) {
+    return ErrorMessage(absl::StrFormat("Unable to load module: %s", elf_file.error().message()));
+  }
+
+  ErrorMessageOr<uint64_t> load_bias = elf_file.value()->GetLoadBias();
+  // Every loadable module contains a load bias.
+  if (!load_bias) {
+    return ErrorMessage(absl::StrFormat("No load bias found for module %s", module_path));
```

Comment:
I suggest `: %s` or `\"%s\"` like above.

Reply:
changed to use load_bias.error() here.